mica-core/src/main/java/org/obiba/mica/dataset/service/HarmonizationDatasetService.java

```diff
@@ -141,6 +142,11 @@ public HarmonizationDataset findById(@NotNull String id) throws NoSuchDatasetExc
     return dataset;
   }
 
+  public List<HarmonizationDataset> findAllDatasetsByHarmonizationStudy(@NotNull String studyId) {
+    Assert.isTrue(!Strings.isNullOrEmpty("Harmonization Study ID cannot be null or empty"));
```

Comment:
Euh ... do you test if "Harmonization Study ID cannot be null or empty" is null or empty ?
Did you test your code ? ...